projects/baby_driver/scripts/gpio_interrupts.py

```diff
@@ -0,0 +1,182 @@
+        edge = getattr(InterruptEdge, edge.upper())
+
+    if (edge < 0 or edge >= InterruptEdge.NUM_INTERRUPT_EDGES):
+        raise ValueError("invalid interrupt edge (enter 0 (Rising), 1 (Falling) \
+                        or 2 (Rising_and_falling")
+
+    if callback is not None and (callable(callback)) is False:
+        raise ValueError("invalid callback function")
+
+    msg_data_register_gpio_interrupt = [(port, 1), (pin, 1), (edge, 1)]
+
+    can_util.send_message(
+        babydriver_id = message_defs.BabydriverMessageId.GPIO_IT_REGISTER_COMMAND,
+        data = can_util.can_pack(msg_data_register_gpio_interrupt)
+    )
+
+    status_message = can_util.next_message(babydriver_id=message_defs.BabydriverMessageId.STATUS)
+    received_status = status_message.data[1]
+
+    # Check if status is invalid (0 refers to STATUS_CODE_OK)
+    if received_status != 0:
+        raise Exception("received a nonzero STATUS_CODE: {}".format(received_status))
+
+    # Adding callback to dictionary to be called upon when interrupt occurs
+    if callback is None:
+        callback_dict[(port, pin)] = default_callback
+    else:
+        callback_dict[(port, pin)] = callback
+
+
+
+def unregister_gpio_interrupt(port, pin):
+    """
+    Unregisters a gpio interrupt on a pin
+
+    Args:
+        port: Port of the GPIO pin to register an interrupt in
+        pin: Pin number of the GPIO pin to register an interrupt in
+
+    Raises:
+        Value error: if the parameters passed into register_gpio_interrupt are incorrect
+        Attribute error: if the port parameter is invalid (refer gpio_port.py
+                         for acceptable port parameters)
+        Exception: if a non-zero status code is received when attempting to uregister an interrupt
+        Key error: if no interrupt is registered in the given port and pin
+    """
+
+    if isinstance(port, str):
+        port = getattr(GpioPort, port.capitalize())
+
+    if port < 0 or port >= GpioPort.NUM_GPIO_PORTS:
+        raise ValueError("invalid GPIO port (Range: 'A' - 'F')")
+
+    if pin < 0 or pin >= NUM_PINS_PER_PORT:
+        raise ValueError("Invalid GPIO pin number (Range: 0 - 15)")
+
+    msg_data_unregister_gpio_interrupt = [(port, 1), (pin, 1)]
+
+    can_util.send_message(
+        babydriver_id = message_defs.BabydriverMessageId.GPIO_IT_UNREGISTER_COMMAND,
+        data = can_util.can_pack(msg_data_unregister_gpio_interrupt)
+    )
+
+    status_message = can_util.next_message(babydriver_id=message_defs.BabydriverMessageId.STATUS)
+    received_status = status_message.data[1]
+
+    # Check if status is invalid (0 refers to STATUS_CODE_OK)
+    if received_status != 0:
+        raise Exception("Received a nonzero STATUS_CODE: {}".format(received_status))
+
+    # Clearing callback related to the interrupt that was unregistered
+    if (port, pin) not in callback_dict:
+        raise KeyError(f"No interrupt registered on given port and pin {callback_dict}")
```

Comment:
I can't guarantee that all the Python versions we'll run this on are 3.6+ and support f-strings, so could you replace all your f-strings with `.format` instead?